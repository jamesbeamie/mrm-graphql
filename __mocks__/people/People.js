const allPeopleReturnData = {
  data: {
    users: {
      users: [
        {
          email: 'michael.mukalo@andela.com',
          name: 'Michael Mukalo',
          picture: 'https://lh4.googleusercontent.com/-nvDSOOB7tuI/AAAAAAAAAAI/AAAAAAAAABA/otjsk3lztyc/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '5',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'marlone.akidiva@andela.com',
          name: 'Marlone Akidiva',
          picture: 'https://lh4.googleusercontent.com/-2TgU0EHynu8/AAAAAAAAAAI/AAAAAAAAAAc/ML95I5e2k_c/photo.jpg?sz=50',
          location: 'Nairobi',
          id: '6',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
        {
          email: 'george.tibetegya@andela.com',
          name: 'George Tibetegya',
          picture: 'https://lh3.googleusercontent.com/-yhiUccQX5FE/AAAAAAAAAAI/AAAAAAAAAAc/fsBnDfr2fu8/photo.jpg?sz=50',
          location: 'Kampala',
          id: '7',
          roles: [
            {
              role: 'Admin',
              id: '2',
              __typename: 'Role',
            },
          ],
          __typename: 'User',
        },
      ],
      __typename: 'Users',
      pages: 2,
      queryTotal: 11,
      hasNext: true,
      hasPrevious: false,
    },
  },
};

export { allPeopleReturnData as default };
