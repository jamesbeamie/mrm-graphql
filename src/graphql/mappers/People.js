/**
 * This file will have functions to format data returned by queries in the backend
 * to fit into the current implementation of components while doing less work.
 *
 * I have done this because of the changing requirements and data layer of our backend
 * We stand a risk of changing our components whenever the backend changes
 *
 * In my opinion, instead of always modifying our components when the backend changes
 * We can just reformat our data before we pass it to the component.
 *
 * By doing this, we remain with the current implementation of components,
 * no change in snapshots and all we need to do is to change the data mapping methods
 */

export const formatPeopleData = person => ({
  id: person.id,
  name: person.name,
  email: person.email,
  location: person.location,
  accessLevel: person.roles[0].role,
  picture: person.picture,
});

export { formatPeopleData as default };
