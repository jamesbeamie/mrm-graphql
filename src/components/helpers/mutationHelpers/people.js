/* eslint-disable import/prefer-default-export */
import apolloClient from '../../../utils/ApolloClient';
import { INVITE_PERSON_MUTATION, CHANGE_USER_LOCATION_MUTATION } from '../../../graphql/mutations/People';
import { decodeTokenAndGetUserData } from '../../../utils/Cookie';
import { GET_USER_QUERY } from '../../../graphql/queries/People';
import GET_ALL_LEVELS from '../../../graphql/queries/Levels';
import { GET_DEVICES_QUERY } from '../../../graphql/queries/Devices';


const invitePersonMutation = async (email, client = apolloClient) => {
  await client
    .mutate({
      mutation: INVITE_PERSON_MUTATION,
      name: 'invitePersonMutation',
      variables: { email },
    });
};

const changeUserLocation = async (locationId, queryEmail, client = apolloClient) => {
  const { UserInfo: { email: tokenEmail } } = decodeTokenAndGetUserData();
  const email = queryEmail || tokenEmail;

  const { data } = await client
    .mutate({
      mutation: CHANGE_USER_LOCATION_MUTATION,
      name: 'changeUserLocation',
      variables: { email, locationId },
      refetchQueries: [
        { query: GET_USER_QUERY, variables: { email } },
        { query: GET_ALL_LEVELS },
        { query: GET_DEVICES_QUERY },
      ],
    });

  return data;
};

export { invitePersonMutation, changeUserLocation };
