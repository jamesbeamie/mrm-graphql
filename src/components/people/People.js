import React from 'react';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { ApolloConsumer } from 'react-apollo';
import { IconMenu, MenuItem } from 'react-toolbox/lib/menu';
import notification from '../../utils/notification';
import { getAllLocationsFromCache } from '../helpers/QueriesHelpers';
import { changeUserLocation } from '../helpers/mutationHelpers/people';
import { defaultUserIcon } from '../../utils/images/images';

/**
 * Shows the menuCart besides the Access Level on the people's table
 *
 * @returns {JSX}
 */
const accessMenuCaret = () => <span className="access-by-caret" />;

/**
 * Component that shows people and their attributes
 *
 * @param {Object} people - name, email, accessLevel, location, and picture
 * @param {Object} allRoles
 * @param {function} editRole
 *
 * @returns {void}
 */
const People = ({
  people: {
    email, name, accessLevel, location, picture,
  },
  allRoles, perPage, refetch, currentPage, editRole,
}) => {
  const editLocationFunction = async (locId, emailOfUser) => {
    const response = await changeUserLocation(locId, emailOfUser);
    notification(
      toastr,
      'success',
      `${response.changeUserLocation.user.name}'s location has been changed to ${response.changeUserLocation.user.location}`,
    )();
    refetch({ page: currentPage, perPage });
  };

  const editRoleFunction = (roleId) => {
    const variables = { variables: { email, roleId } };
    editRole(variables)
      .then(() => {
        notification(
          toastr,
          'success',
          `'${name}' role has been changed successfully`,
        )();
        refetch({ page: currentPage, perPage });
      })
      .catch((err) => {
        notification(toastr, 'error', err.graphQLErrors[0].message)();
      });
  };
  return (
    <ApolloConsumer>
      {
        client => (
          <div className="table__row">
            <span>
              <img
                className="profilePic"
                id={email}
                alt="alt"
                onError={() => {
                  const element = document.getElementById(email);
                  if (element.alt === 'alt') element.src = defaultUserIcon;
                  element.alt = 'profile';
                }}
                src={picture}
              />
              {name}
            </span>
            <span>{location}
              <IconMenu
                position="topRight"
                className="people-access-dropdown"
                icon={accessMenuCaret()}
              >
                {getAllLocationsFromCache().map(loc => (
                  <MenuItem
                    className={`access-menu ${
                      loc.name === location ? 'selected' : ''
                      }`}
                    key={loc.name}
                    caption={loc.name}
                    onClick={() => {
                      client.writeData({ data: { userLocation: loc } });
                      editLocationFunction(loc.id, email);
                    }
                    }
                  />
                ))}
              </IconMenu>
            </span>
            <span>
              <span>
                {accessLevel}
                <IconMenu
                  position="topRight"
                  className="people-access-dropdown"
                  icon={accessMenuCaret()}
                >
                  {allRoles.map(role => (
                    <MenuItem
                      className={`access-menu ${
                        role.role === accessLevel ? 'selected' : ''
                        }`}
                      key={role.id}
                      caption={role.role}
                      onClick={() => editRoleFunction(role.id)}
                    />
                  ))}
                </IconMenu>
              </span>
            </span>
          </div>
        )}
    </ApolloConsumer>);
};

People.propTypes = {
  people: PropTypes.shape({
    email: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    location: PropTypes.string,
    accessLevel: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
  }).isRequired,
  allRoles: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      role: PropTypes.string,
    }),
  ).isRequired,
  editRole: PropTypes.func.isRequired,
  refetch: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  perPage: PropTypes.number.isRequired,
};

export default People;
