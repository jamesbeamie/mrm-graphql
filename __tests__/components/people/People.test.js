import React from 'react';
import { mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';
import { ApolloError } from 'apollo-client';
import People from '../../../src/components/people/People';
import allLocations from '../../../__mocks__/offices/Locations';
import * as queryHelper from '../../../src/components/helpers/QueriesHelpers';
import * as mutationHelper from '../../../src/components/helpers/mutationHelpers/people';

jest.mock('../../../src/components/helpers/QueriesHelpers');
jest.mock('../../../src/components/helpers/mutationHelpers/people');

queryHelper.getAllLocationsFromCache.mockReturnValue(allLocations.data.allLocations);

describe('People Component', () => {
  const result = { data: { email: 'test@email.com', id: 1 } };
  const person = {
    name: 'Edward Barton',
    email: 'edward.barton@andela.com',
    location: 'Nairobi',
    accessLevel: 'Admin',
    picture: 'http://no-url',
  };
  const allRoles = [
    { id: 1, role: 'Admin' },
    { id: 2, role: 'Default User' },
  ];
  const editRole = jest.fn(() => Promise.resolve(result));
  const refetch = jest.fn(() => Promise.resolve(result));

  let spy;

  const wrapper = mount(
    <MockedProvider addTypename={false}>
      <People
        allLocations={allLocations}
        people={person}
        refetch={refetch}
        allRoles={allRoles}
        editRole={editRole}
        currentPage={1}
        perPage={3}
      />
    </MockedProvider>,
  );


  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain role menu', async () => {
    const iconBtn = wrapper.find('ThemedIconMenu');
    expect(iconBtn.length).toEqual(2);
    iconBtn.last().simulate('click');
    const iconMenu = wrapper.find('ThemedThemed');
    expect(iconMenu.length).toEqual(7);
  });

  it('should change role succesfully', () => {
    const iconMenu = wrapper.find('ThemedThemed').at(6);
    iconMenu.simulate('click');
    expect(editRole).toBeCalled();
    expect(editRole).toBeCalledWith({ variables: { email: person.email, roleId: 2 } });
  });

  it('should call change location function', () => {
    const iconMenu = wrapper.find('ThemedThemed').at(3);
    iconMenu.simulate('click');
    spy = jest.spyOn(mutationHelper, 'changeUserLocation').mockImplementationOnce(() => ({}));
    expect(spy).toBeCalled();
    expect(refetch).toBeCalled();
  });

  it('should return an error when changing role', () => {
    wrapper.setProps({
      editRole: jest.fn(() => Promise.reject(new ApolloError(
        { graphQLErrors: [new Error('error')] }))),
    });
    const iconMenu = wrapper.find('ThemedThemed').at(6);
    iconMenu.simulate('click');
    expect(editRole).toBeCalled();
    expect(editRole).toBeCalledWith({ variables: { email: person.email, roleId: 2 } });
  });
});
