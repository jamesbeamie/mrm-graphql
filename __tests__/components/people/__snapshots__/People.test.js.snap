// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Component renders properly 1`] = `
<MockedProvider
  addTypename={false}
>
  <ApolloProvider
    client={
      ApolloClient {
        "__operations_cache__": Map {},
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {},
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "dataStore": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "fetchQueryRejectFns": Map {},
          "idCounter": 1,
          "inFlightLinkObservables": Map {},
          "link": MockLink {
            "addTypename": false,
            "mockedResponsesByKey": Object {},
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutationStore": MutationStore {
            "store": Object {},
          },
          "onBroadcast": [Function],
          "pollingInfoByQueryId": Map {},
          "queries": Map {},
          "queryDeduplication": true,
          "queryStore": QueryStore {
            "store": Object {},
          },
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "typeDefs": undefined,
        "version": "2.6.4",
        "watchQuery": [Function],
      }
    }
  >
    <People
      allLocations={
        Object {
          "data": Object {
            "allLocations": Array [
              Object {
                "id": "2",
                "name": "Kampala",
              },
              Object {
                "id": "3",
                "name": "Nairobi",
              },
              Object {
                "id": "1",
                "name": "Lagos",
              },
            ],
          },
        }
      }
      allRoles={
        Array [
          Object {
            "id": 1,
            "role": "Admin",
          },
          Object {
            "id": 2,
            "role": "Default User",
          },
        ]
      }
      currentPage={1}
      editRole={[MockFunction]}
      people={
        Object {
          "accessLevel": "Admin",
          "email": "edward.barton@andela.com",
          "location": "Nairobi",
          "name": "Edward Barton",
          "picture": "http://no-url",
        }
      }
      perPage={3}
      refetch={[MockFunction]}
    >
      <ApolloConsumer>
        <div
          className="table__row"
        >
          <span>
            <img
              alt="alt"
              className="profilePic"
              id="edward.barton@andela.com"
              onError={[Function]}
              src="http://no-url"
            />
            Edward Barton
          </span>
          <span>
            Nairobi
            <ThemedIconMenu
              className="people-access-dropdown"
              composeTheme="deeply"
              icon={
                <span
                  className="access-by-caret"
                />
              }
              iconRipple={true}
              mapThemrProps={[Function]}
              menuRipple={true}
              position="topRight"
              selectable={false}
            >
              <IconMenu
                className="people-access-dropdown"
                icon={
                  <span
                    className="access-by-caret"
                  />
                }
                iconRipple={true}
                menuRipple={true}
                position="topRight"
                selectable={false}
                theme={Object {}}
              >
                <div
                  className="people-access-dropdown"
                >
                  <ThemedThemed
                    composeTheme="deeply"
                    disabled={false}
                    icon={
                      <span
                        className="access-by-caret"
                      />
                    }
                    mapThemrProps={[Function]}
                    onClick={[Function]}
                    ripple={true}
                    rippleCentered={true}
                    rippleClassName=""
                    rippleMultiple={true}
                    rippleSpread={2}
                  >
                    <ThemedRippledComponent
                      composeTheme="deeply"
                      disabled={false}
                      icon={
                        <span
                          className="access-by-caret"
                        />
                      }
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={true}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                      theme={Object {}}
                    >
                      <RippledComponent
                        disabled={false}
                        icon={
                          <span
                            className="access-by-caret"
                          />
                        }
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={true}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                      >
                        <IconButton
                          accent={false}
                          className=""
                          disabled={false}
                          icon={
                            <span
                              className="access-by-caret"
                            />
                          }
                          neutral={true}
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          primary={false}
                          theme={Object {}}
                          type="button"
                        >
                          <button
                            className=""
                            data-react-toolbox="button"
                            disabled={false}
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchStart={[Function]}
                            type="button"
                          >
                            <span
                              className="access-by-caret"
                            />
                          </button>
                        </IconButton>
                      </RippledComponent>
                    </ThemedRippledComponent>
                  </ThemedThemed>
                  <ThemedMenu
                    active={false}
                    composeTheme="deeply"
                    mapThemrProps={[Function]}
                    onHide={[Function]}
                    outline={true}
                    position="topRight"
                    ripple={true}
                    selectable={false}
                    theme={Object {}}
                  >
                    <Menu
                      active={false}
                      onHide={[Function]}
                      outline={true}
                      position="topRight"
                      ripple={true}
                      selectable={false}
                      theme={Object {}}
                    >
                      <div
                        className=""
                        data-react-toolbox="menu"
                        style={
                          Object {
                            "height": undefined,
                            "width": undefined,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "height": undefined,
                              "width": undefined,
                            }
                          }
                        />
                        <ul>
                          <ThemedThemed
                            caption="Kampala"
                            className="access-menu "
                            composeTheme="deeply"
                            disabled={false}
                            key=".$Kampala"
                            mapThemrProps={[Function]}
                            onClick={[Function]}
                            ripple={true}
                            rippleCentered={false}
                            rippleClassName=""
                            rippleMultiple={true}
                            rippleSpread={2}
                            selected={false}
                          >
                            <ThemedRippledComponent
                              caption="Kampala"
                              className="access-menu "
                              composeTheme="deeply"
                              disabled={false}
                              mapThemrProps={[Function]}
                              onClick={[Function]}
                              ripple={true}
                              rippleCentered={false}
                              rippleClassName=""
                              rippleMultiple={true}
                              rippleSpread={2}
                              selected={false}
                              theme={Object {}}
                            >
                              <RippledComponent
                                caption="Kampala"
                                className="access-menu "
                                disabled={false}
                                onClick={[Function]}
                                ripple={true}
                                rippleCentered={false}
                                rippleClassName=""
                                rippleMultiple={true}
                                rippleSpread={2}
                                selected={false}
                                theme={Object {}}
                              >
                                <MenuItem
                                  caption="Kampala"
                                  className="access-menu "
                                  disabled={false}
                                  onClick={[Function]}
                                  onMouseDown={[Function]}
                                  onTouchStart={[Function]}
                                  selected={false}
                                  theme={Object {}}
                                >
                                  <li
                                    className="access-menu "
                                    data-react-toolbox="menu-item"
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onTouchStart={[Function]}
                                  >
                                    <span>
                                      Kampala
                                    </span>
                                  </li>
                                </MenuItem>
                              </RippledComponent>
                            </ThemedRippledComponent>
                          </ThemedThemed>
                          <ThemedThemed
                            caption="Nairobi"
                            className="access-menu selected"
                            composeTheme="deeply"
                            disabled={false}
                            key=".$Nairobi"
                            mapThemrProps={[Function]}
                            onClick={[Function]}
                            ripple={true}
                            rippleCentered={false}
                            rippleClassName=""
                            rippleMultiple={true}
                            rippleSpread={2}
                            selected={false}
                          >
                            <ThemedRippledComponent
                              caption="Nairobi"
                              className="access-menu selected"
                              composeTheme="deeply"
                              disabled={false}
                              mapThemrProps={[Function]}
                              onClick={[Function]}
                              ripple={true}
                              rippleCentered={false}
                              rippleClassName=""
                              rippleMultiple={true}
                              rippleSpread={2}
                              selected={false}
                              theme={Object {}}
                            >
                              <RippledComponent
                                caption="Nairobi"
                                className="access-menu selected"
                                disabled={false}
                                onClick={[Function]}
                                ripple={true}
                                rippleCentered={false}
                                rippleClassName=""
                                rippleMultiple={true}
                                rippleSpread={2}
                                selected={false}
                                theme={Object {}}
                              >
                                <MenuItem
                                  caption="Nairobi"
                                  className="access-menu selected"
                                  disabled={false}
                                  onClick={[Function]}
                                  onMouseDown={[Function]}
                                  onTouchStart={[Function]}
                                  selected={false}
                                  theme={Object {}}
                                >
                                  <li
                                    className="access-menu selected"
                                    data-react-toolbox="menu-item"
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onTouchStart={[Function]}
                                  >
                                    <span>
                                      Nairobi
                                    </span>
                                  </li>
                                </MenuItem>
                              </RippledComponent>
                            </ThemedRippledComponent>
                          </ThemedThemed>
                          <ThemedThemed
                            caption="Lagos"
                            className="access-menu "
                            composeTheme="deeply"
                            disabled={false}
                            key=".$Lagos"
                            mapThemrProps={[Function]}
                            onClick={[Function]}
                            ripple={true}
                            rippleCentered={false}
                            rippleClassName=""
                            rippleMultiple={true}
                            rippleSpread={2}
                            selected={false}
                          >
                            <ThemedRippledComponent
                              caption="Lagos"
                              className="access-menu "
                              composeTheme="deeply"
                              disabled={false}
                              mapThemrProps={[Function]}
                              onClick={[Function]}
                              ripple={true}
                              rippleCentered={false}
                              rippleClassName=""
                              rippleMultiple={true}
                              rippleSpread={2}
                              selected={false}
                              theme={Object {}}
                            >
                              <RippledComponent
                                caption="Lagos"
                                className="access-menu "
                                disabled={false}
                                onClick={[Function]}
                                ripple={true}
                                rippleCentered={false}
                                rippleClassName=""
                                rippleMultiple={true}
                                rippleSpread={2}
                                selected={false}
                                theme={Object {}}
                              >
                                <MenuItem
                                  caption="Lagos"
                                  className="access-menu "
                                  disabled={false}
                                  onClick={[Function]}
                                  onMouseDown={[Function]}
                                  onTouchStart={[Function]}
                                  selected={false}
                                  theme={Object {}}
                                >
                                  <li
                                    className="access-menu "
                                    data-react-toolbox="menu-item"
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onTouchStart={[Function]}
                                  >
                                    <span>
                                      Lagos
                                    </span>
                                  </li>
                                </MenuItem>
                              </RippledComponent>
                            </ThemedRippledComponent>
                          </ThemedThemed>
                        </ul>
                      </div>
                    </Menu>
                  </ThemedMenu>
                </div>
              </IconMenu>
            </ThemedIconMenu>
          </span>
          <span>
            <span>
              Admin
              <ThemedIconMenu
                className="people-access-dropdown"
                composeTheme="deeply"
                icon={
                  <span
                    className="access-by-caret"
                  />
                }
                iconRipple={true}
                mapThemrProps={[Function]}
                menuRipple={true}
                position="topRight"
                selectable={false}
              >
                <IconMenu
                  className="people-access-dropdown"
                  icon={
                    <span
                      className="access-by-caret"
                    />
                  }
                  iconRipple={true}
                  menuRipple={true}
                  position="topRight"
                  selectable={false}
                  theme={Object {}}
                >
                  <div
                    className="people-access-dropdown"
                  >
                    <ThemedThemed
                      composeTheme="deeply"
                      disabled={false}
                      icon={
                        <span
                          className="access-by-caret"
                        />
                      }
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={true}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                    >
                      <ThemedRippledComponent
                        composeTheme="deeply"
                        disabled={false}
                        icon={
                          <span
                            className="access-by-caret"
                          />
                        }
                        mapThemrProps={[Function]}
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={true}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                      >
                        <RippledComponent
                          disabled={false}
                          icon={
                            <span
                              className="access-by-caret"
                            />
                          }
                          onClick={[Function]}
                          ripple={true}
                          rippleCentered={true}
                          rippleClassName=""
                          rippleMultiple={true}
                          rippleSpread={2}
                          theme={Object {}}
                        >
                          <IconButton
                            accent={false}
                            className=""
                            disabled={false}
                            icon={
                              <span
                                className="access-by-caret"
                              />
                            }
                            neutral={true}
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            primary={false}
                            theme={Object {}}
                            type="button"
                          >
                            <button
                              className=""
                              data-react-toolbox="button"
                              disabled={false}
                              onClick={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchStart={[Function]}
                              type="button"
                            >
                              <span
                                className="access-by-caret"
                              />
                            </button>
                          </IconButton>
                        </RippledComponent>
                      </ThemedRippledComponent>
                    </ThemedThemed>
                    <ThemedMenu
                      active={false}
                      composeTheme="deeply"
                      mapThemrProps={[Function]}
                      onHide={[Function]}
                      outline={true}
                      position="topRight"
                      ripple={true}
                      selectable={false}
                      theme={Object {}}
                    >
                      <Menu
                        active={false}
                        onHide={[Function]}
                        outline={true}
                        position="topRight"
                        ripple={true}
                        selectable={false}
                        theme={Object {}}
                      >
                        <div
                          className=""
                          data-react-toolbox="menu"
                          style={
                            Object {
                              "height": undefined,
                              "width": undefined,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "height": undefined,
                                "width": undefined,
                              }
                            }
                          />
                          <ul>
                            <ThemedThemed
                              caption="Admin"
                              className="access-menu selected"
                              composeTheme="deeply"
                              disabled={false}
                              key=".$1"
                              mapThemrProps={[Function]}
                              onClick={[Function]}
                              ripple={true}
                              rippleCentered={false}
                              rippleClassName=""
                              rippleMultiple={true}
                              rippleSpread={2}
                              selected={false}
                            >
                              <ThemedRippledComponent
                                caption="Admin"
                                className="access-menu selected"
                                composeTheme="deeply"
                                disabled={false}
                                mapThemrProps={[Function]}
                                onClick={[Function]}
                                ripple={true}
                                rippleCentered={false}
                                rippleClassName=""
                                rippleMultiple={true}
                                rippleSpread={2}
                                selected={false}
                                theme={Object {}}
                              >
                                <RippledComponent
                                  caption="Admin"
                                  className="access-menu selected"
                                  disabled={false}
                                  onClick={[Function]}
                                  ripple={true}
                                  rippleCentered={false}
                                  rippleClassName=""
                                  rippleMultiple={true}
                                  rippleSpread={2}
                                  selected={false}
                                  theme={Object {}}
                                >
                                  <MenuItem
                                    caption="Admin"
                                    className="access-menu selected"
                                    disabled={false}
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onTouchStart={[Function]}
                                    selected={false}
                                    theme={Object {}}
                                  >
                                    <li
                                      className="access-menu selected"
                                      data-react-toolbox="menu-item"
                                      onClick={[Function]}
                                      onMouseDown={[Function]}
                                      onTouchStart={[Function]}
                                    >
                                      <span>
                                        Admin
                                      </span>
                                    </li>
                                  </MenuItem>
                                </RippledComponent>
                              </ThemedRippledComponent>
                            </ThemedThemed>
                            <ThemedThemed
                              caption="Default User"
                              className="access-menu "
                              composeTheme="deeply"
                              disabled={false}
                              key=".$2"
                              mapThemrProps={[Function]}
                              onClick={[Function]}
                              ripple={true}
                              rippleCentered={false}
                              rippleClassName=""
                              rippleMultiple={true}
                              rippleSpread={2}
                              selected={false}
                            >
                              <ThemedRippledComponent
                                caption="Default User"
                                className="access-menu "
                                composeTheme="deeply"
                                disabled={false}
                                mapThemrProps={[Function]}
                                onClick={[Function]}
                                ripple={true}
                                rippleCentered={false}
                                rippleClassName=""
                                rippleMultiple={true}
                                rippleSpread={2}
                                selected={false}
                                theme={Object {}}
                              >
                                <RippledComponent
                                  caption="Default User"
                                  className="access-menu "
                                  disabled={false}
                                  onClick={[Function]}
                                  ripple={true}
                                  rippleCentered={false}
                                  rippleClassName=""
                                  rippleMultiple={true}
                                  rippleSpread={2}
                                  selected={false}
                                  theme={Object {}}
                                >
                                  <MenuItem
                                    caption="Default User"
                                    className="access-menu "
                                    disabled={false}
                                    onClick={[Function]}
                                    onMouseDown={[Function]}
                                    onTouchStart={[Function]}
                                    selected={false}
                                    theme={Object {}}
                                  >
                                    <li
                                      className="access-menu "
                                      data-react-toolbox="menu-item"
                                      onClick={[Function]}
                                      onMouseDown={[Function]}
                                      onTouchStart={[Function]}
                                    >
                                      <span>
                                        Default User
                                      </span>
                                    </li>
                                  </MenuItem>
                                </RippledComponent>
                              </ThemedRippledComponent>
                            </ThemedThemed>
                          </ul>
                        </div>
                      </Menu>
                    </ThemedMenu>
                  </div>
                </IconMenu>
              </ThemedIconMenu>
            </span>
          </span>
        </div>
      </ApolloConsumer>
    </People>
  </ApolloProvider>
</MockedProvider>
`;
